### PR TITLE
add selector sync

### PR DESCRIPTION
**Pull request by hermesthecat**

### CODE REVIEW

File: options.js [hunk 1]
```diff
@@ -2,217 +2,281 @@
...
-    document.body.appendChild(dl);
-    dl.click();
-    document.body.removeChild(dl);
+const impbtnWrp = document.getElementById("impbtn_wrapper");
+const impbtn = document.getElementById("impbtn");
+const expbtn = document.getElementById("expbtn");
+
+// Sync button click event
+syncButton.addEventListener("click", function () {
+  await browser.storage.sync.get("selectors", function (result) {
```
Comment (igorlogius) [inline options.js:181]: no callback please

File: options.js [hunk 2]
```diff
@@ -2,217 +2,281 @@
...
+  await browser.storage.local.get("selectors", function (localResult) {
+    if (localResult.selectors) {
+      localResult.selectors.forEach((selector) => {
+        selector.action = "delete";
         createTableRow(selector);
-    });
+      });
+    } else {
+      // If no local selectors, check for synced selectors
+     await browser.storage.sync.get("selectors", function (syncResult) {
```
Comment (igorlogius) [inline options.js:157]: no callback please

File: options.js [hunk 3]
```diff
@@ -2,217 +2,281 @@
...
+  bookmarkFolders = await browser.runtime.sendMessage({});
+
+  createTableRow({
+    activ: 1,
+    bookmarkId: "",
+    url_regex: "",
+    action: "save",
+  });
+
+  //let res = await browser.storage.local.get('selectors');
```
Comment (igorlogius) [inline options.js:142]: no code in comments necessary, can be removed

**Review by igorlogius — CHANGES_REQUESTED** [review]

File: options.js [hunk 4]
```diff
@@ -2,217 +2,281 @@
...
+browser.storage.local.get("syncEnabled", function (result) {
+  if (result.syncEnabled !== undefined) {
+    syncCheckbox.checked = result.syncEnabled;
+  }
+});
+
 function deleteRow(rowTr) {
-    let mainTableBody = document.getElementById('mainTableBody');
-    mainTableBody.removeChild(rowTr);
+  let mainTableBody = document.getElementById("mainTableBody");
```
[the remark this answers is not in the document]
Comment (hermesthecat) [reply options.js:16]: not important for me, can stay as before.

**Comment by igorlogius** [comment]
> approve or reject, but I don't have time to deal with it any more.

well, with your consent i'm gonna finish the PR and restructure the necessary elements. 

Thanks for submiting the idea and this inital draft. :+1:

**Comment by igorlogius** [comment]
Had to make a few more adjustments but ... now a user can make a backup that can be pulled down from any device. An auto-sync options seemed to be a bit more then necessary especially since it always comes with the risk of merge issues/expectations, which i really dont want to deal with. Better to leave that up to the user. 

Sorry if the changes now do not align with your expectations, but i guess you can now just as well maintain your forked version.  

Anyway, thanks for the PRs and conversation and feel free to submit other ideas if you like. 

Have a nice day.

**Pull request closed without merging**